spec/roo_on_rails/harness_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+require 'roo_on_rails/harness'
+require 'roo_on_rails/checks/environment'
+
+RSpec.describe RooOnRails::Harness do
+  describe '#run' do
+    let(:try_fix) { false }
+    let(:context_value) { {} }
+    let(:checks_class) { RooOnRails::Checks::Environment }
+    let(:staging_check) { instance_double(checks_class, run: true) }
+    let(:production_check) { instance_double(checks_class, run: true) }
+    let(:instance) {described_class.new(try_fix: try_fix, context: context_value) }
+
+    subject(:run) { instance.run }
+
+    before do
+      allow(checks_class).to receive(:new) do |options|
```

Comment:
Stubbing `new` on classes (that are implementation detail) is a bit iffy... Any chance to revisit this a bit?
An idea could be, for instance, to create a "dry run" mode for checks, and have the harness return/expose the context so we can check what was attempted.